dubbo-common/src/main/java/org/apache/dubbo/common/utils/ClassHelper.java

```diff
@@ -265,26 +266,38 @@ public static boolean isPrimitive(Class<?> type) {
     }
 
     public static Object convertPrimitive(Class<?> type, String value) {
-        if (type == char.class || type == Character.class) {
+        if (value == null) {
+            return null;
+        } else if (type == char.class || type == Character.class) {
             return value.length() > 0 ? value.charAt(0) : '\0';
         } else if (type == boolean.class || type == Boolean.class) {
             return Boolean.valueOf(value);
         } else if (type == byte.class || type == Byte.class) {
-            return Byte.valueOf(value);
+            return isNumber(value) ? Byte.valueOf(value) : null;
```

Comment:
I don't think it is a good way to use regular express to check whether it is a number before doing the conversion, which is quite expensive. Please take a look at org.apache.dubbo.common.utils.StringUtils#isNumeric()

Reply:
This method cannot exclude numbers with decimal points.

Reply:
Maybe you can enhance it based on the logic provided in `org.apache.dubbo.common.utils.StringUtils#isNumeric()`

Reply:
thanks,I will modify it.